### PR TITLE
[improvement] Listview widget - Adds default children when dropped for the first time

### LINKED ISSUE #3616: Add default widgets to Listview
### Description
With default data, the **Listview** widget should have an image, a text and a button. 

For example, if the length of the default data is 3 then the child components for the listview widget will be 3 rows, each row with the 3 widgets

### PR DESCRIPTION
**Pull request by arpitnath**
Resolves #3616

### CODE REVIEW

**Comment by arpitnath** [comment]
Todo: Update the base branch.
Dependent PR: #3715

File: frontend/src/_hooks/use-mount.jsx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+
+export const useMounted = () => {
```
Comment (sherfin94) [inline frontend/src/_hooks/use-mount.jsx:3]: Hey @arpitnath , could you explain how this `useMounted` hook ensures that the child widgets are dropped only once?

To be more specific, how is this different from using `useEffect` with an empty dependency array?

**Review by sherfin94 — CHANGES_REQUESTED** [review]
Great refactoring!: not a single line needs to be changed inside the ListView component 🔥 

Few issues:
1. child widgets are not created when I drop listView inside a container
2. When list view is dropped,  the debugger shows a very vague validation error, can debug this with you on call if needed.

Also asked for explanation of the `useMount` hook.

File: frontend/src/_hooks/use-mount.jsx (hunk 1 above)
Comment (arpitnath) [reply frontend/src/_hooks/use-mount.jsx:3]: We can only add new children once the `appDefinations` are updated. as during the first render cycle, the component (listview) mounts but the `subcontainer` mounts after it and the appDefinations is updated as soon as the widget is dropped.
As the parent mounts first and the promise is waiting to get resolved and update the state(editor), right away if we check the children which will be an empty array but on adding the children to the boxes(subcontainer layer), the app breaks. 
Here we are trying to compute components's state which is waiting to be resolved and updated resulting in accessors as undefined (component.component.name and component.component.exposedVariables)

p.s. It took me quite some time to figure out these co-relations.

**Comment by Nandinisaha13** [comment]
<html>
<body>
<google-sheets-html-origin>Test Cases

Test Case | Steps | Resullt
-- | -- | --
List View should be able to be dragged & drop on canvas. | Create an application. | Application gets created.
  | Drag and drop list view. | List view is displayed on the canvas.
Should be able to change the list view's name. | Create an application. | Application gets created.
  | Drag and drop list view. | List view is displayed on the canvas.
  | Click on list view. | The properties is displayed on the right.
  | Enter name. | Name is changed.
Should be able to change the list data. | Create an application. | Application gets created.
  | Drag and drop list view. | List view is displayed on the canvas.
  | Click on list view. | The properties is displayed on the right.
  | Enter the changes on list data. | The changes are reflected on canvas.
Should be able to change the row height. | Create an application. | Application gets created.
  | Drag and drop list view. | List view is displayed on the canvas.
  | Click on list view. | The properties is displayed on the right.
  | Enter height in row height field. | The row height is increased/decreased.
Should be able to hide the bottom border. | Create an application. | Application gets created.
  | Drag and drop list view. | List view is displayed on the canvas.
  | Click on list view. | The properties is displayed on the right.
Should be able to show alert. | Create an application. | Application gets created.
  | Drag and drop list view. | List view is displayed on the canvas.
  | Click on add event handler. | The event handler is added.
  | Click on a row of list view. | Alert is shown.
Should be able to have a list view inside a list view.(5 level depth i.e listview inside listview inside listview inside listview inside listview) | Create an application. | Application gets created.
  | Drag and drop listview. | Listview is displayed on the canvas.
  | Drag and drop another listview inside the previous listview. | Listview inside listview is displayed.
Should be able to drop any other widget inside list view. | Create an application. | Application gets created.
  | Drag and drop listview. | Listview is displayed on the canvas.
  | Drag and drop widgets on first row. | The widgets will then be shown in every row.
Should be able to view default widgets in list view. | Create an application. | Application gets created.
  | Drag and drop list view. | List view has button, text & image as default.
Should be able to add new widgets while default widget already present. | Create an application. | Application gets created.
  | Drag and drop list view. | List view has button, text & image as default.
  | Drag and drop widgets on first row. | The widgets are added to the list view.
Should be able to remove the default widgets & add new widgets. | Create an application. | Application gets created.
  | Drag and drop list view. | List view has button, text & image as default.
  | Delete the default widgets. | The default widgets are deleted.
  | Drag and drop widgets on first row of list view. | The widgets are added to the list view.
Should be able to delete the default widgets & add list view. | Create an application. | Application gets created.
  | Drag and drop list view. | List view has button, text & image as default.
  | Delete the default widgets. | The default widgets are deleted.
  | Drag and drop list view on first row. | The list view is added.
  | Delete the default widgets in next list view. | The default widgets are deleted.
  | Drag and drop list view on first row. | The list view is added & nesting of list view works.
Should be able to remove one/two default widgets and add list view. | Create an application. | Application gets created.
  | Drag and drop list view. | List view has button, text & image as default.
  | Delete one/two default widgets. | The default widgets are deleted.
  | Drag and drop another list view on the first row of the previous list view. | The new list view list view is added to the previous list view & nesting works.
Should be able to delete list view. | Create an application. | Application gets created.
  | Drag and drop list view on the canvas. | List view is added to the canvas.
  | Select the list view. | List view is selected.
  | Click on delete icon. | List view is deleted.
Should be able to give some border to list view. | Create an application. | Application gets created.
  | Drag and drop list view. | List view is added to the canvas.
  | Select the list view. | List view is selected.
  | Go to styles. | Border field is displayed.
  | Enter 30 in border. | Border is rounded accordingly.
Should be able to give box shadow. | Create an application. | Application gets created.
  | Drag and drop list view. | List view is added to the canvas.
  | Select the list view. | List view is selected.
  | Go to styles. | General is displayed.
  | Click on general. | Box shadow field is displayed.
  | Change X,Y, Blur,Spread accordingly or click on fx and then change the options. | The changes for box shadow is displayed.
Should be able to give clear the box shadow settings. | Create an application. | Application gets created.
  | Drag and drop list view. | List view is added to the canvas.
  | Select the list view. | List view is selected.
  | Go to styles. | General is displayed.
  | Click on general. | Box shadow field is displayed.
  | Change X,Y, Blur,Spread accordingly or click on fx and then change the options. | The changes for box shadow is displayed.
  | Click on box shadow. | The panel opens up.
  | Click on clear button. | The box shadow are reset.
Should be able to change color for background. | Create an application. | Application gets created.
  | Drag and drop list view. | List view is added to the canvas.
  | Select the list view. | List view is selected.
  | Go to styles. | Background Color is displayed.
  | Change the color or click on fx and enter the hex code for color. | The background color changes.
Should be able to not view list view. | Create an application. | Application gets created.
  | Drag and drop list view. | List view is added to the canvas.
  | Go to styles. | Visible toggle is displayed.
  | Toggle the visibility field or click on fx and enter false. | The list view is not visible.
Should be able to disable the list view. | Create an application. | Application gets created.
  | Drag & drop list view. | List view is added to the canvas.
  | Go to styles. | Disable toggle is displayed.
  | Toggle the Disable field or click on fx and enter true. | The list view is disabled.
Should be able to display default widgets while listview is inside a container. | Create an application. | Application gets created.
  | Drag & drop container. | Container is added to the canvas.
  | Drag & drop list view. | List view is added to the container.
  | Observe. | Default widgets are present on list view.
Should be able to delete default widgets while list view is inside a container. | Create an application. | Application gets created.
  | Drag & drop container. | Container is added to the canvas.
  | Drag & drop list view. | List view is added to the container.
  | Click on one default widgets. | Widget is selected.
  | Click on delete icon. | Widget is deleted.
Should be able to drop any other widget inside list view while in container. | Create an application. | Application gets created.
  | Drag & drop container. | Container is added to the canvas.
  | Drag and drop listview. | Listview is displayed on the container.
  | Drag and drop widgets on first row of list view. | The widgets will then be shown in every row.
Should be able to remove the default widgets & add new widgets in list view while inside container. | Create an application. | Application gets created.
  | Drag & drop container. | Container is added to the canvas.
  | Drag and drop list view. | List view has button, text & image as default on container.
  | Delete the default widgets. | The default widgets are deleted.
  | Drag and drop widgets on first row of list view. | The widgets are added to the list view.
Should be able to display default widgets while listview is inside a tab. | Create an application. | Application gets created.
  | Drag & drop tab. | Tab is added to the canvas.
  | Drag & drop list view. | List view is added to the container.
  | Observe. | Default widgets are present on list view.
Should be able to delete default widgets while list view is inside a tab. | Create an application. | Application gets created.
  | Drag & drop tab. | Tab is added to the canvas.
  | Drag & drop list view. | List view is added to the container.
  | Click on one default widgets. | Widget is selected.
  | Click on delete icon. | Widget is deleted.
Should be able to drop any other widget inside list view while in tab. | Create an application. | Application gets created.
  | Drag & drop tab. | Tab is added to the canvas.
  | Drag and drop listview. | Listview is displayed on the container.
  | Drag and drop widgets on first row of list view. | The widgets will then be shown in every row.
Should be able to remove the default widgets & add new widgets in list view while inside tab. | Create an application. | Application gets created.
  | Drag & drop tab. | Tab is added to the canvas.
  | Drag and drop list view. | List view has button, text & image as default on tab.
  | Delete the default widgets. | The default widgets are deleted.
  | Drag and drop widgets on first row of list view. | The widgets are added to the list view.
Should be able to display default widgets while listview is inside a modal. | Create an application. | Application gets created.
  | Drag & drop modal & button. | Button & modal is added to the canvas.
  | Create an event as action type show modal & select the modal name. | Event is created.
  | Click on button. | Modal is opened
  | Drag & drop list view. | List view is added to the modal.
  | Observe. | Default widgets are present on list view.
Should be able to delete default widgets while list view is inside a modal. | Create an application. | Application gets created.
  | Drag & drop modal & button. | Button & modal is added to the canvas.
  | Create an event as action type show modal & select the modal name. | Event is created.
  | Click on button. | Modal is opened
  | Drag & drop list view. | List view is added to the container.
  | Click on one default widgets. | Widget is selected.
  | Click on delete icon. | Widget is deleted.
Should be able to drop any other widget inside list view while in modal. | Create an application. | Application gets created.
  | Drag & drop modal & button. | Button & modal is added to the canvas.
  | Create an event as action type show modal & select the modal name. | Event is created.
  | Click on button. | Modal is opened
  | Drag and drop listview. | Listview is displayed on the container.
  | Drag and drop widgets on first row of list view. | The widgets will then be shown in every row.
Should be able to remove the default widgets & add new widgets in list view while inside modal. | Create an application. | Application gets created.
  | Drag & drop modal & button. | Button & modal is added to the canvas.
  | Create an event as action type show modal & select the modal name. | Event is created.
  | Click on button. | Modal is opened
  | Drag and drop list view. | List view has button, text & image as default on container.
  | Delete the default widgets. | The default widgets are deleted.
  | Drag and drop widgets on first row of list view. | The widgets are added to the list view.


</body>
</html>

**Review by sherfin94 — APPROVED** [review]

**Comment by shubh22** [comment]
Task linked: [CU-2namw25 Default widgets for Container layout widgets](https://app.clickup.com/t/2namw25)

**Pull request merged**